Makefile

```
@@ -74,6 +76,15 @@ vet: ## Run go vet against code.
 .PHONY: test
 test: manifests generate fmt vet envtest ## Run tests.
 	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
+	grep -v -e "_moq.go" cover.out > coverage.out
+
+test/output/coverage/report:
+	@if [ -f coverage.out ]; then $(GO) tool cover -func=coverage.out; else echo "coverage.out file not found"; fi;
+.PHONY: test/output/coverage/report
+
+test/html/coverage/report:
+	@if [ -f coverage.out ]; then $(GO) tool cover -html=coverage.out; else echo "coverage.out file not found"; fi;
+.PHONY: test/html/coverage/report
```

```suggestion

coverage.out: test

coverage-report: coverage.out
	$(GO) tool cover -func=coverage.out
.PHONY: coverage-report

html-coverage-report: coverage.out
	$(GO) tool cover -html=coverage.out
.PHONY: html-coverage-report
```

WDYT about something like this.

---

You can leverage make dependencies so that you don't need to do a check to see if the file exists, and if it doesn't exist it will automatically be created.

---

Nice suggestion, thanks @KPostOffice